### DISCUSSION TITLE
How to always use tabs?

### DISCUSSION THREAD

**Discussion by wnhrt** [neovim/neovim#30550] [Q&A] [answered]
I am aware of `expandtab` and disabled it in my configuration.
`:verbose set expandtab?` tells me in some files, e.g. markdown, that it is overwritten by the neovim runtime ftplugin.
Is there any way to disable this, without manualy creating the same configuration for each file type?
There is no case in which I want `expandtab` enabled automatically.

**Comment by wnhrt** [comment, marked as answer]
This problem bugged me on and off for a while now and it took only 15 more minutes to find a solution...

```lua
-- ./init.lua
vim.b.did_ftplugin = 1
```

**Comment by telemachus** [comment]
`vim.b.did_ftplugin = 1` will probably prevent all filetype plugins from doing anything. That's giving up **a lot** of functionality.

> Is there any way to disable this, without manualy creating the same configuration for each file type?

Yeah, it's a drag, but it doesn't have to be the end of the world.

```shell
mkdir -p your/dotfiles/config/nvim/after/ftplugin
echo "vim.bo.expandtab = false" > your/dotfiles/config/nvim/after/ftplugin/sh.lua
# alternatively...
# echo "setlocal noexpandtab" > your/dotfiles/config/nvim/after/ftplugin/sh.vim
for i in awk c css html mail markdown perl python tex text vim # ...
do
cp -v your/dotfiles/config/nvim/after/ftplugin/sh.lua your/dotfiles/config/nvim/after/ftplugin/${i}.lua
# alternatively...
#  cp -v your/dotfiles/config/nvim/after/ftplugin/sh.vim your/dotfiles/config/nvim/after/ftplugin/${i}.vim
done
```

**Reply by zeertzjq** [reply]
There are also variables like `g:markdown_recommended_style` that can be before loading ftplugins.
